data/data/openstack/masters/private-network.tf

```diff
@@ -1,9 +1,12 @@
 locals {
   # Create subnet for the first MachineNetwork CIDR if we need to
   nodes_cidr_block = var.machine_v4_cidrs[0]
-  nodes_subnet_id  = var.openstack_machines_subnet_id != "" ? var.openstack_machines_subnet_id : openstack_networking_subnet_v2.nodes[0].id
-  nodes_network_id = var.openstack_machines_network_id != "" ? var.openstack_machines_network_id : openstack_networking_network_v2.openshift-private[0].id
-  create_router    = (var.openstack_external_network != "" && var.openstack_machines_subnet_id == "") ? 1 : 0
+  nodes_default_port = var.openstack_default_machines_port != null ? var.openstack_default_machines_port : {
+    network_id = openstack_networking_network_v2.openshift-private[0].id,
+    fixed_ips  = [{ subnet_id = openstack_networking_subnet_v2.nodes[0].id, ip_address = "" }],
+  }
+  nodes_ports   = [for port in var.openstack_machines_ports : port != null ? port : local.nodes_default_port]
+  create_router = (var.openstack_external_network != "" && var.openstack_default_machines_port == null) ? 1 : 0
```

Comment:
I have removed two values that we used to pass two values to Terraform:
* `openstack_machines_network_id` was the network of `machinesSubnet`, or empty
* `openstack_machines_subnet_id` was `machinesSubnet`, or it was empty

If empty, Terraform created a new network.

Now the logic is similar, but I have started using this new `port` structure instead. The reason for this change is that we need to be able to pass multiple subnets per network. This is irrelevant for machinesSubnet and for the Installer-generated network, but it's essential when we have a control-plane portTarget. In that case, we want to be able to define a port with multiple subnets (most notably, for IPv6).

Reply:
The old variables are replaced with:
* `openstack_default_machines_port`: machinesSubnet, or null
* `openstack_machines_ports`: array of length(number-of-masters) defining the primary port for the master. If null, it's going to be the network created by Terraform.

Here we replace the null values, if present, with data from the network created by Terraform.